crates/oxc_linter/src/rules/jest/no_large_snapshots.rs

```diff
@@ -0,0 +1,554 @@
+            self.report_in_span(span, ctx);
+        }
+    }
+
+    fn report_in_expr_stmt(&self, expr_stmt: &ExpressionStatement, ctx: &LintContext) {
+        self.check_all_paths_are_absolute();
+
+        let line_count = Self::get_line_count(expr_stmt.span, ctx);
+        let allowed = match &expr_stmt.expression {
+            Expression::AssignmentExpression(assignment_expr) => {
+                let Some(member_expr) = assignment_expr.left.as_member_expression() else {
+                    return;
+                };
+                self.check_allowed_in_snapshots(member_expr, ctx)
+            }
+            Expression::BinaryExpression(binary_expr) => {
+                let Some(member_expr) = binary_expr.left.as_member_expression() else {
+                    return;
+                };
+                self.check_allowed_in_snapshots(member_expr, ctx)
+            }
+            Expression::LogicalExpression(logical_expr) => {
+                let Some(member_expr) = logical_expr.left.as_member_expression() else {
+                    return;
+                };
+                self.check_allowed_in_snapshots(member_expr, ctx)
+            }
+            _ => false,
+        };
+
+        if !allowed && line_count > self.max_size {
+            if line_count == 0 {
+                ctx.diagnostic(no_snapshot(line_count, expr_stmt.span));
+            } else {
+                ctx.diagnostic(too_long_snapshots(self.max_size, line_count, expr_stmt.span));
+            }
+        }
+    }
+
+    fn report_in_span(&self, span: Span, ctx: &LintContext) {
+        self.check_all_paths_are_absolute();
+
+        let line_count = Self::get_line_count(span, ctx);
+
+        if line_count > self.inline_max_size {
+            if self.inline_max_size == 0 {
+                ctx.diagnostic(no_snapshot(line_count, span));
+            } else {
+                ctx.diagnostic(too_long_snapshots(self.inline_max_size, line_count, span));
+            }
+        }
+    }
+
+    fn check_allowed_in_snapshots(
+        &self,
+        member_expr: &MemberExpression,
+        ctx: &LintContext,
+    ) -> bool {
+        let Some(snapshot_name) = member_expr.static_property_name() else {
+            return false;
+        };
+        let Some(file_name) = ctx.file_path().to_str() else {
+            return false;
+        };
+
+        let Some(allowed_snapshots_in_file) = self.allowed_snapshots.get(file_name) else {
+            return false;
+        };
+
+        allowed_snapshots_in_file.iter().any(|allowed_snapshot| {
+            match Regex::new(allowed_snapshot) {
+                Ok(regex) => regex.is_match(snapshot_name),
+                Err(_) => snapshot_name == allowed_snapshot,
+            }
+        })
+    }
+
+    fn check_all_paths_are_absolute(&self) {
+        let all_paths_are_absolute =
+            self.allowed_snapshots.keys().all(|p| Path::new(p).is_absolute());
+
+        assert!(all_paths_are_absolute,
```

Comment:
Would you like to remove  `assert` here?(I noticed the `throw Error` in jest plugin:) )

Reply:
Done.